### ISSUE TITLE
Not using merged arrays from config file

### ISSUE THREAD

**Issue by theoribeiro** [aheissenberger/serverless-appsync-offline#27]
This plugin is not using the full potential of appsync-emulator or appsync-plugin when it comes to merging an array of files in Mapping Templates. For example, if I have the following configuration: 
```
  mappingTemplates:
    - ${file(notification-templates.yml)}
    - ${file(user-templates.yml)}
```
Appsync-emulator will resolve it properly and so will appsync-plugin. This is done by the loadServerlessConfig function in Appsync-emulator that is ignored when appsync-offline builds the server since it calls createServer passing this.serverless, instead of passing the path to the Serverless config file. 

Here is how appsync-emulator calls it's own createServer function on `bin/server.js`
```
const serverless = path.join(path.dirname(pkgPath), 'serverless.yml');
const server = await createServer({ wsPort, serverless, port, dynamodb });
```

Because of this call, appsync-emulator calls loadServerlessConfig:
```
  if (typeof serverless === 'object') {
    serverlessConfig = serverless.service;
    serverlessDirectory = serverless.config.servicePath;
  } else {
    serverlessDirectory =
      typeof serverless === 'string'
        ? path.dirname(serverless)
        : findServerlessPath();
    const config = await loadServerlessConfig(serverlessDirectory); // This is called
    serverlessConfig = config.config;
  }
```

loadServerlessConfig does the flattening of the arrays:
```
const output = {
    config: {
      ...config,
      custom: {
        ...custom,
        appSync: {
          ...appSync,
          mappingTemplates: flatten(mappingTemplates),
          dataSources: flatten(dataSources),
        },
      },
      resources: normalizeResources(config),
    },
    directory: cwd,
  };
```

In order to avoid having to do the same in appsync-offline, I suggest simply calling appsync-emulator's createServer in the same way it calls itself. This would support the use case that both the other two plugins in this ecosystem support. 

Does all this make sense? I can create a PR if you are willing to take this suggestion. 

--Theo

**Comment by aheissenberger** [comment]
@theobribeiro I am happy to accept a PR which keeps this plugin in sync with AppSync emulator

**Comment by aheissenberger** [comment]
@theoribeiro I still would appreciate to accept a PR for this change as you have a good test case :-)

**Comment by aheissenberger** [comment]
@theoribeiro  I did look into this and do not see your problem. All Serverless variables are resolved by the Serverless framework before a hook is called:
https://serverless.com/framework/docs/providers/aws/guide/plugins/#serverless-instance

I tested & debugged this with the following configuration:

serverless.yml
```yml
    mappingTemplates:
      - dataSource: NotesTableDS
        type: Query
        field: allNotes
        request: "allnotes-request.vtl"
        response: "allnotes-response.vtl"
      - ${file(mappingTemplateGetNote.yml)}
```

mappingTemplateGetNote.yml
```yml
dataSource: NotesTableDS
type: Query
field: getNote
request: "getnote-request.vtl"
response: "common-response.vtl"
```

when I start local with:
`sls offline`

everything is resolved.

Using `https://github.com/ConduitVC/aws-utils/blob/1149ebdf2e73d15d358afab1abe39957aa115073/packages/appsync-emulator-serverless/loadServerlessConfig.js` is not an option as it is a hack for the use case of using the emulator without the Serverless framework.

**Comment by inventionlabsSydney** [comment]
@aheissenberger Hi,
This problem exists when you want to split into multiple files but those files have multiple mappings in them, see here: https://github.com/sid88in/serverless-appsync-plugin/issues/265

I have isolated it to: https://github.com/ConduitVC/aws-utils/issues/86 

I think the best course of action is to update this package.json to use 0.14.2 of @conduitvc/appsync-emulator-serverless

Would that be plausible?

**Comment by aheissenberger** [comment]
the package is defined as "^0.14.0" which will install any version >=0.14.0 - you will get this version without any change of package.json by calling "yarn upgrade"

**Comment by LMulvey** [comment]
To piggy back off of this, you're technically right @aheissenberger but that change is reflected version `1.4.0` of this package, which has not been published to npm. So, the current version (`1.3.3`) still pulls in a version of `appsync-emulator-serverless` that does not have this fix. 

@inventionlabsSydney  The fix until `1.4.0` is published is to point your dependency to the GitHub repository in your `package.json` file and run `npm install` again. This will grab version `1.4.5` of `appsync-emulator-serverless`, which resolves this issue.

```
"serverless-appsync-offline": "aheissenberger/serverless-appsync-offline",
```

I've opened #42 to address the publishing issue.
